### PR TITLE
[PM-15547] Revoke managed user on 2FA removal if enforced by organization policy

### PR DESCRIPTION
**Pull request by r-tome**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-15547

## 📔 Objective

When the two factor organization policy is enabled, then user should be revoked when they remove their 2FA provider if the policy applies to them.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Core/AdminConsole/Enums/EventSystemUser.cs [hunk 1]
```diff
@@ -6,4 +6,5 @@ public enum EventSystemUser : byte
     SCIM = 1,
     DomainVerification = 2,
     PublicApi = 3,
+    TwoFactorDisabling = 4,
```
Comment (r-tome) [inline src/Core/AdminConsole/Enums/EventSystemUser.cs:9]: I'm not sure about this name, I welcome suggestions

Comment (JimmyVo16) [reply src/Core/AdminConsole/Enums/EventSystemUser.cs:9]: Maybe `TwoFactorDisabled`? `Disabled` sounds more like a `state`, but I don't feel strongly about this. The name you choose isn't a blocker for me.

File: src/Core/Services/Implementations/UserService.cs [hunk 2]
```diff
@@ -1355,13 +1361,27 @@ public void SetTwoFactorProvider(User user, TwoFactorProviderType type, bool set
...
         var twoFactorPolicies = await _policyService.GetPoliciesApplicableToUserAsync(user.Id, PolicyType.TwoFactorAuthentication);
+        var organizationsManagingUser = await GetOrganizationsManagingUserAsync(user.Id);
 
         var removeOrgUserTasks = twoFactorPolicies.Select(async p =>
         {
-            await _removeOrganizationUserCommand.RemoveUserAsync(p.OrganizationId, user.Id);
             var organization = await _organizationRepository.GetByIdAsync(p.OrganizationId);
-            await _mailService.SendOrganizationUserRemovedForPolicyTwoStepEmailAsync(
-                organization.DisplayName(), user.Email);
+            if (_featureService.IsEnabled(FeatureFlagKeys.AccountDeprovisioning) && organizationsManagingUser.Any(o => o.Id == p.OrganizationId))
```
Comment (JimmyVo16) [inline src/Core/Services/Implementations/UserService.cs:1369]: I was thinking of moving organizationsManagingUser into the if statement so it's only called if the feature check is on, but your way is cleaner, and if performance isn't an issue, I think it's the better option.

**Review by JimmyVo16 — DISMISSED** [review]

File: src/Core/Services/Implementations/UserService.cs (hunk 2 above)
Comment (r-tome) [reply src/Core/Services/Implementations/UserService.cs:1369]: If the feature flag is disabled then `GetOrganizationsManagingUserAsync` simply returns an empty list.
ref: https://github.com/bitwarden/server/blob/170836aba1f87ee4a5c5a0b33d76dfa487ee94d1/src/Core/Services/Implementations/UserService.cs#L1295

File: src/Core/AdminConsole/Enums/EventSystemUser.cs (hunk 1 above)
Comment (r-tome) [reply src/Core/AdminConsole/Enums/EventSystemUser.cs:9]: Makes sense to me, I'll rename it

**Pull request merged**